packages/@aws-cdk/aws-rds/lib/props.ts

```diff
@@ -144,14 +135,55 @@ export interface CredentialsFromUsernameOptions {
   readonly excludeCharacters?: string;
 }
 
+/**
+ * Options for creating Credentials from a username.
+ */
+export interface CredentialsFromUsernameOptions extends CredentialsBaseOptions {
+  /**
+   * Password
+   *
+   * Do not put passwords in your CDK code directly.
+   *
+   * @default - a Secrets Manager generated password
+   */
+  readonly password?: SecretValue;
+}
+
 /**
  * Username and password combination
  */
 export abstract class Credentials {
+  /**
+   * Creates Credentials with a password generated and stored in SecretsManager.
+   */
+  public static fromGeneratedPassword(username: string, options: CredentialsBaseOptions = {}): Credentials {
+    return {
+      ...options,
+      username,
+      usernameAsString: true,
+    };
+  }
+
+  /**
+   * Creates Credentials from a password
+   *
+   * Do not put passwords in your CDK code directly.
+   */
+  public static fromPassword(username: string, password: SecretValue): Credentials {
+    return {
+      username,
+      password,
+      usernameAsString: true,
+    };
+  }
 
   /**
    * Creates Credentials for the given username, and optional password and key.
    * If no password is provided, one will be generated and stored in SecretsManager.
+   *
+   * @deprecated use `fromGeneratedPassword()` or `fromPassword()` for new deployments only.
+   *   Switching to `fromGeneratedPassword()` or `fromPassword()` for existing instances/clusters
+   *   will have them replaced.
```

Comment:
I feel like it's important we're very precise with the wording here.

```suggestion
   * @deprecated use `fromGeneratedPassword()` or `fromPassword()` for new Clusters and Instances.
   *   Note that switching from `fromUsername()` to `fromGeneratedPassword()` or `fromPassword()` for already deployed
   *  Clusters or Instances will result in their replacement!
```